### DISCUSSION TITLE
How to automatically open database with key file stored in attachment?

### DISCUSSION THREAD

**Discussion by mpeter50** [keepassxreboot/keepassxc#9159] [Q&A] [answered]
I was following the instructions [written here](https://keepassxc.org/docs/KeePassXC_UserGuide.html#_automatic_database_opening) for making my database open another one when it is unlocked.

I see that this feature supports opening databases with key files, but that requires writing the path for the file in the username field.
What are my options if I store the key file as an attachment of this auto-open entry? Is this possible to do? I haven't found any information on how could I reference attachments.
If only externally stored key files are supported, that seems counterintuitive, as those are not encrypted, and are available to any other process.

I have found #6947 which is the first half of what I am looking for, but that was not implemented.

**Comment by droidmonkey** [comment, marked as answer]
Why do you need a key file at all? Just use a long complex password. It is not possible to use an attachment.
